runelite-client/src/main/java/net/runelite/client/plugins/screenshot/ScreenshotPlugin.java

```diff
@@ -362,6 +377,31 @@ public void hideWidgets(WidgetHiddenChanged event)
 		takeScreenshot(fileName, config.displayDate());
 	}
 
+	@Subscribe
+	private  void onGameStateChange(GameStateChanged gameStateChanged)
+	{
+		int currentWorld = client.getWorld();
+		try
+		{
+			World world = worldResult.findWorld(currentWorld);
+			if (world.getTypes().size() >= 1)
+			{
+				String worldTypeName = world.getTypes().iterator().next().name();
+				IS_DMM_WORLD = worldTypeName.equals(WorldType.DEADMAN.name());
+				IS_DMM_WORLD = !IS_DMM_WORLD ? worldTypeName.equals(WorldType.SEASONAL_DEADMAN.name()) : true;
+			}
+			else
+			{
+				IS_DMM_WORLD = false;
+			}
+		}
```

Comment:
I think everything inside the try { } can be replaced with:

```java
World world = worldResult.findWorld(currentWorld);

boolean deadman = world.getTypes().contains(WorldType.DEADMAN);
boolean seasonalDeadman = world.getTypes().contains(WorldType.SEASONAL_DEADMAN);

IS_DMM_WORLD = deadman || seasonalDeadman;
```

Reply:
Rewrote it with your code and seem to be working so I'll update it.